### ISSUE TITLE
Some thoughts on improving community and collaboration

### ISSUE THREAD

**Issue by hawkw** [hawkw/alarm#42]
Since ALARM now has a few contributors other than me, I've been thinking about working toward improving community aspects of the project. As there are currently only a few of us, some of these things might be a bit premature, but I wanted to start soliciting input from you all now.

Here are some things I've been thinking about:

+ creating an SOS GitHub organisation and moving both ALARM and [the kernel repository](https://github.com/hawkw/sos-kernel) into its namespace
   + this would let me manage contributors to both repos in one place, so that contributors to the kernel repo are also ALARM contributors and vice versa. This isn't terribly important right now but might be later, as we start actually integrating ALARM allocators with the kernel.
  + allow us to create GitHub projects spanning issues from both repos --- this is something I'd like to be able to do, as there are a number of kernel projects that will also involve work in this repository.
  + give us access to [GitHub team discussions](https://github.com/blog/2471-introducing-team-discussions) which would be useful for talking about planning and design in ways that don't directly correspond to single issues
+ creating other venues for communication, like a Slack, Mattermost, or Gitter chat channel. 
  + i'm not sure if this is something we need or not, but it might be nice to have a less formal, real-time way to discuss things and ask questions.

I'm tagging everyone who's contributed to this repo so far (@amanjeev, @croyzor, @jdanford, @leshow, @croyzor), plus everyone who's made significant contributions to the kernel repo (@MxLinux, @rachlmac, and @twisted-pear), because I'd love to hear your thoughts on this.

**Comment by amanjeev** [comment]
👍 for both suggestions. In fact, I am really interested in the other repo as well. 😃 

I have not used Mattermost but use Slack and Gitter extensively. I find Slack to be much better but that is my personal opinion.

**Comment by hawkw** [comment]
Thanks @amanjeev! I totally agree that Slack is a much better product than Gitter, but Slack is invitation-only, which makes it less useful as a venue for people to ask questions or get help using the library. On the third hand, I haven't actually _used_ Mattermost, but am somewhat attracted to the fact that it's open source.

**Comment by amanjeev** [comment]
> On the third hand, I haven't actually used Mattermost, but am somewhat attracted to the fact that it's open source.

Ooo I did not know! So, given that this is an informal real time chat stuff, I say we can give Mattermost a try and experiment how it feels. What do you say?

> but Slack is invitation-only

hmm... I thought it depends on how to setup Slack up. I believe you could make it open to anyone to sign up for your Slack org. Isn't it? Or am I mistaken about it.

**Comment by twisted-pear** [comment]
I don't have strong feelings about the github org. Whatever is easiest and best for you Eliza works for me.
I'm ok with Slack or Mattermost, but I think if this is also supposed to be used as a channel for getting support, the easier to join the better.

**Comment by hawkw** [comment]
> I'm ok with Slack or Mattermost, but I think if this is also supposed to be used as a channel for getting support, the easier to join the better.

Yeah, I'm somewhat unsure at the moment what the primary purpose of a real-time chat would be --- is it more for dev discussions, or for support and help? Do we want one chat platform with separate channels for both? Obviously support isn't an issue right now, but it would be cool if it was someday...

**Comment by hawkw** [comment]
@amanjeev 
> hmm... I thought it depends on how to setup Slack up. I believe you could make it open to anyone to sign up for your Slack org. Isn't it? Or am I mistaken about it.

AFAIK, it's invitation-only but there are apps like [Slackin](https://github.com/rauchg/slackin) where people can input an email and it automatically sends you an invitation. I'd rather not have to set up hosting just to run something like that, though.

**Comment by twisted-pear** [comment]
Support could also mean "How can I make it compile?" or "I'm looking at the code and have zero clue what to do". Not everyone who takes a look is gonna become a contributor, but if they get some help getting started the chances are a lot better I think.

**Comment by hawkw** [comment]
> Support could also mean "How can I make it compile?" or "I'm looking at the code and have zero clue what to do". Not everyone who takes a look is gonna become a contributor, but if they get some help getting started the chances are a lot better I think.

Definitely. I suspect that as we continue building the kernel we'll get a lot of questions of this nature, and I'd like to have a venue for answering them with as low a barrier to entry as possible. I'd _also_ like to have a place to discuss dev things in depth, and I'd like these to be in the _same_ place, so we don't have to say something like, "okay, time has come for you to be blessed into A Contributor, now you have to join this completely separate chat platform".

**Comment by leshow** [comment]
I'm good with all of that, I saw this repo after you tweeted some open issues on twitter. I've been using rust for a while but don't have a lot of experience with low level projects and thought this would be fun, so whatever helps you organize things better is awesome in my books.

**Comment by hawkw** [comment]
I've created the [SOS](https://github.com/sos-os)  GitHub org, moved this repo & the kernel repo into the org, and invited y'all to join. Started organizing some tickets into cross-repository projects, as well.

I'm going to close this issue for now, though let's keep discussing collaboration & communication moving forward.
